webknossos/webknossos/dataset/dataset.py

```diff
@@ -630,11 +632,17 @@ def from_images(
         if use_bioformats is not False:
             valid_suffixes.update(pims_images.get_valid_bioformats_suffixes())
 
-        input_files = [
-            i.relative_to(input_upath)
-            for i in input_upath.glob("**/*")
-            if i.is_file() and i.suffix.lstrip(".").lower() in valid_suffixes
-        ]
+        if input_upath.is_dir():
```

Comment:
On non-local file systems, is_file is significantly cheaper than is-dir. Can you flip the if/else block?